packages/samples/headless-ssr/src/components/result-list.tsx

```diff
@@ -1,18 +1,39 @@
-'use client';
+import {
+  ResultList as ResultListController,
+  ResultListState,
+} from '@coveo/headless';
+import {useEffect, useState, FunctionComponent} from 'react';
 
-import {Result} from '@coveo/headless';
+interface ResultListProps {
+  initialState: ResultListState;
+  controller?: ResultListController;
+}
+
+export const ResultList: FunctionComponent<ResultListProps> = (props) => {
+  const {initialState, controller} = props;
+  const [state, setState] = useState(initialState);
+
+  useEffect(
+    () => controller?.subscribe(() => setState({...controller.state})),
```

Comment:
AFAIK, normally the controller should be a dependency specifically because it's being used?

Specifically, we want the effect to be re-started whenever the `controller` pointer's value changes (e.g., when the engine finishes hydrating and it goes from `undefined` to a value).

If the `controller` changes for any reason, we also want to unsubscribe from the old controller, which is handled here by returning the unsubscribe function returned by `subscribe`.

I'll add e2e tests.